opal/datatype/opal_convertor_internal.h

```diff
@@ -22,10 +23,255 @@
 
 BEGIN_C_DECLS
 
-typedef int32_t (*conversion_fct_t)( opal_convertor_t* pConvertor, uint32_t count,
-                                     const void* from, size_t from_len, ptrdiff_t from_extent,
-                                     void* to, size_t to_length, ptrdiff_t to_extent,
-                                     ptrdiff_t *advance );
+#define COPY_TO_VECTOR   1
+#define COPY_FROM_VECTOR 2
+typedef size_t (*conversion_fct_t)( opal_convertor_t* pConvertor, int mode,
+                                     char* vector_buf,
+                                     size_t vector_len,
+                                     size_t nblocks,
+                                     size_t count_per_block,
+                                     size_t extent_between_blocks,
+                                     size_t vector_element_size,
+                                     int vector_is_bigendian,
```

Comment:
not sure why you need the 2 is_bigendian, as one of the sides of the pack/unpack operation is always local.

Reply:
The main reason for that change is packing an 8 byte MPI_LONG down to the 4 bytes specified by external32.  Previously it was only packing 8 bytes into 8 bytes, and in that case you don't need to know which buffer is which, just knowing they differ in endianness is enough to swap the bytes around.  But if you're packing 8 bytes down to 4 you need to know which 4 bytes to pick.  For that you need to know if "from" is big endian and "to" is little endian or vice versa.

However now that I've switched the arguments specifying the two buffers to be
1. a vector and
2. a packed buf
I agree that argument became redundant because the vector is always local with its endianness specified in opal_local_arch.